### PR TITLE
feat: add metrics configuration support to the Dapr SDK

### LINKED ISSUE #1281: Add Configuration support for metrics
## Expected Behavior

https://docs.dapr.io/operations/configuration/configuration-overview/#metrics

https://github.com/dapr/java-sdk/blob/master/testcontainers-dapr/src/main/java/io/dapr/testcontainers/Configuration.java#L19

## Release Note

RELEASE NOTE:

### PR DESCRIPTION
**Pull request by matheusandre1**
# Description

Adds support for the `spec.metrics` section of the Dapr Configuration to the `testcontainers-dapr` module, following the same pattern used for the recently added `mtls` and `logging` settings.

New classes in `io.dapr.testcontainers`:

* `MetricsConfigurationSettings` – `enabled`, `rules`, `latencyDistributionBuckets`, `http` and `recordErrorCodes`.
* `HttpMetricsConfigurationSettings` – `increasedCardinality`, `pathMatching` and `excludeVerbs`.
* `MetricsRule` and `MetricsLabel` – named rules with label/regex filters.

Changes:

* `Configuration` gets a new `metrics` field, a new constructor overload and a getter. Existing constructors are kept, so this is backward compatible.
* `ConfigurationYamlConverter` renders the `metrics` block, omitting null fields and empty lists (same behavior as the mTLS and logging blocks).
* `ConfigurationYamlConverterTest` covers full, minimal, HTTP-only, empty-list, rule-without-labels and combined logging+metrics configurations.

Example:

```java
MetricsConfigurationSettings metrics = new MetricsConfigurationSettings(
    true,
    List.of(new MetricsRule("dapr_runtime_service_invocation_req_sent_total",
        List.of(new MetricsLabel("method", Map.of("orders/", "orders/.+"))))),
    List.of(5, 10, 50, 100, 500),
    new HttpMetricsConfigurationSettings(false, List.of("/items", "/orders/{orderID}"), true),
    true
);
```

new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
    .withConfiguration(new Configuration("my-config", null, null, null, null, metrics));

```yaml
spec:
  metrics:
    enabled: true
    rules:
    - name: dapr_runtime_service_invocation_req_sent_total
      labels:
      - name: method
        regex:
          orders/: orders/.+
    latencyDistributionBuckets:
    - 5
    - 10
    - 50
    - 100
    - 500
    http:
      increasedCardinality: false
      pathMatching:
      - /items
      - /orders/{orderID}
      excludeVerbs: true
    recordErrorCodes: true
```

## Issue reference

Closes: #1281 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: testcontainers-dapr/src/main/java/io/dapr/testcontainers/HttpMetricsConfigurationSettings.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   * @param excludeVerbs         if true (default is false), the Dapr HTTP server ignores the request HTTP verb
+   *                             when building the method metric label.
+   */
+  public HttpMetricsConfigurationSettings(
+      Boolean increasedCardinality,
+      List<String> pathMatching,
+      Boolean excludeVerbs
+  ) {
+    this.increasedCardinality = increasedCardinality;
+    this.pathMatching = pathMatching;
```
Comment (javier-aliaga) [inline testcontainers-dapr/src/main/java/io/dapr/testcontainers/HttpMetricsConfigurationSettings.java]: Can we use Collections.unmodifiableList like in MtlsConfigurationSettings.java:91?

File: testcontainers-dapr/src/main/java/io/dapr/testcontainers/MetricsConfigurationSettings.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  public MetricsConfigurationSettings(
+      Boolean enabled,
+      List<MetricsRule> rules,
+      List<Integer> latencyDistributionBuckets,
+      HttpMetricsConfigurationSettings http,
+      Boolean recordErrorCodes
+  ) {
+    this.enabled = enabled;
+    this.rules = rules;
+    this.latencyDistributionBuckets = latencyDistributionBuckets;
```
Comment (javier-aliaga) [inline testcontainers-dapr/src/main/java/io/dapr/testcontainers/MetricsConfigurationSettings.java]: Collections.unmodifiableList ?

File: testcontainers-dapr/src/main/java/io/dapr/testcontainers/MetricsLabel.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  /**
+   * Creates a new metrics label filter.
+   *
+   * @param name  name of the metric label to filter on (e.g. {@code method}).
+   * @param regex map of replacement value to regex expression. Label values matching the regex are replaced by
+   *              the map key (e.g. {@code "orders/" -> "orders/.+"}).
+   */
+  public MetricsLabel(String name, Map<String, String> regex) {
+    this.name = name;
+    this.regex = regex;
```
Comment (javier-aliaga) [inline testcontainers-dapr/src/main/java/io/dapr/testcontainers/MetricsLabel.java]: Collections.unmodifiableMap?

File: testcontainers-dapr/src/main/java/io/dapr/testcontainers/MetricsRule.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  /**
+   * Creates a new metrics rule.
+   *
+   * @param name   name of the metric the rule applies to
+   *               (e.g. {@code dapr_runtime_service_invocation_req_sent_total}).
+   * @param labels labels to filter on, each with a regex expression applied to the metrics path.
+   */
+  public MetricsRule(String name, List<MetricsLabel> labels) {
+    this.name = name;
+    this.labels = labels;
```
Comment (javier-aliaga) [inline testcontainers-dapr/src/main/java/io/dapr/testcontainers/MetricsRule.java]: Collections.unmodifiableList?

**Comment by javier-aliaga** [comment]
Can you add a test with `enabled: null` and also one with `metrics: {}` ?

**Review by javier-aliaga — APPROVED** [review]
